datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -280,6 +323,7 @@ fn date_trunc_coarse(granularity: &str, value: i64) -> Result<i64> {
 fn _date_trunc(
     tu: TimeUnit,
     value: &Option<i64>,
+    tz: Arc<Option<Tz>>,
```

Comment:
I think a Tz is an integer

```rust
    println!("Size of a Tz: {}", std::mem::size_of::<arrow_array::timezone::Tz>());
```

Prints:
```
Size of a Tz: 4
```

So I think we can avoid this Arc. I'll make it a follow on PR